### DISCUSSION TITLE
Expandable card

### DISCUSSION THREAD

**Discussion by PoojMenon** [twilio-labs/paste#1129] [Office hours] [answered]
We're moving the Pricing page to paste and the layout uses cards. Here is the [collection](https://share.goabstract.com/9e67f9ef-6386-4896-954b-5e275320788e). 

Wanted to get general feedback that there is no issue with this approach?

https://user-images.githubusercontent.com/72031986/106197893-24d25c80-61ab-11eb-9880-42c329b05e0f.mov

**Comment by jweintraub** [comment]
Jumping in with a question - this page is very old with a lot of static
information. Are you planning to add all the missing products? If so, how
do you plan to keep this page maintained as we add additional products. It
might be good to consider ways this can be automated with a CMS or
something like that.

Johanna Weintraub
Staff, Product Designer
[image: twilio] <http://www.twilio.com/?utm_source=email_signature>

**Comment by PoojMenon** [comment]
Hi Jo, for this current iteration, the ask was to just move from PHP to Paste - we did not have the bandwidth to do a full overhaul. For the moment, the page will have the same products it does currently.

**Comment by SiTaggart** [comment]
Hi @PoojMenon 

Can I start by asking; Why do you feel it's necessary to partially collapse the content, over showing all the text by default or implementing a fully collapsible [Disclosure](https://paste.twilio.design/components/disclosure#contained-disclosure)?

**Reply by corinne-nagel** [reply]
Hi @PoojMenon , following up here for documentation purposes, what did you decide to go with?

**Reply by PoojMenon** [reply]
Sorry for the delay - I seem to have missed the original notification. 

The reason for this approach was to avoid cards of different sizes on the page. I also didn't want to increase the size of all the cards to accommodate the one card that has lengthier text. 

This is what it currently looks like: 
![image](https://user-images.githubusercontent.com/72031986/106795213-b598b480-6651-11eb-9189-c2a0f8f2a297.png)

**Comment by kate-porter** [comment, marked as answer]
Hey @PoojMenon - I'm not sure what degree of change is in scope for this work, but could you consider updating the page to follow some of the design patterns on the new Explore Products page in Console.zen?

![image](https://user-images.githubusercontent.com/5374045/106826603-79ab2280-6644-11eb-8098-2376f5234fe5.png)

Or, at a minimum, perhaps you can pull some of these shorter product descriptions into this pricing page to eliminate the issue entirely. There is definitely some copy clutter that could be trimmed down here (for example, the Programmable Chat description opens with, "Twilio Programmable Chat allows you to embed..." while the Client description just starts with "Embed..."). More concise, consistent product descriptions will improve the usability of this page, too.

**Reply by kate-porter** [reply]
And one unrelated piece of feedback: since you're doing a full-page overhaul, I'd recommend moving this to the Default theme in Paste rather than using the legacy Console theme, if possible.

**Reply by PoojMenon** [reply]
Hi Katie, thanks for this feedback. Reducing the copy clutter was definitely something I was hoping to do. Unfortunately, when I picked this up last quarter, the ask was just to take the existing page which was in PHP and translate it to Paste. The degree of change thats possible now would be fairly small. We do plan on revisiting the page in the future, so I will keep your comments in mind. 

All the Billing pages are currently using the console theme, but we're in discussions with the engineers to update to the the default theme in the very near future.

**Reply by kate-porter** [reply]
Gotcha. If this is definitely the route y'all need to go, you can accomplish the desired functionality with the Disclosure [primitive](https://paste.twilio.design/primitives/disclosure-primitive).

One thing to consider is that it'd likely be _more_ expensive to implement that functionality than it would be to copy over the product descriptions from the Explore Products page and keep these as static cards. I'd recommend that approach if at all possible, as it removes the unneeded complexity of the "show all" disclosure, and I would imagine the cards changing height could create some funky implications to the rest of the page layout.

**Reply by PoojMenon** [reply]
Thanks Katie. I'll discuss these options with the engineering team.
